week-3/mandatory/cyf-ecommerce-api/homework.js

```diff
@@ -0,0 +1,244 @@
+                  .then((result) => res.json(result.rows))
+                  .catch((e) => console.error(e));
+              });
+              /////////Add a new DELETE endpoint /orders/:orderId to delete an existing order along all the associated order items./////////////////////
+              app.delete("/orders/:order_id", function (req, res) {
+                const orderId = req.params.order_id;
+              
+                pool
+                  .query("DELETE FROM order_items WHERE order_id =$1", [orderId])
+                  .then(() => {
+                    pool
+                      .query("DELETE FROM orders WHERE id=$1", [orderId])
+                      .then(() => res.send(`Customer ${orderId} deleted!`))
+                      .catch((e) => console.error(e));
+                  })
+                  .catch((e) => console.error(e));
+              });
+
+
+
+
+              ////////Add a new DELETE endpoint /customers/:customerId to delete an existing customer only if this customer doesn't have orders.///
+              app.delete("/customers/:customer_id", function (req, res) {
+                const customerId = req.params.customer_id;
+
+                
+                pool
+                 .query("SELECT * FROM orders WHERE customer_id =$1", [customerId])
+  
+                .then((result) => {
+                    if (result.rows.length > 0) {
+                      return res
+                        .status(400)
+                        .send("This customer has order!");
+                    } else {
+
+                       pool 
+                        .query("DELETE FROM customers WHERE id =$1", [customerId])
+                        .then(() => res.send(`Customer ${customerId} deleted!`))
+                        .catch((e) => console.error(e));
+                    }
+                  });
+              });
+////Add a new GET endpoint /customers/:customerId/orders to load all the orders along the items in the orders of a specific customer. Especially, the following information should be returned: order references, order dates, product names, unit prices, suppliers and quantities.////
+              app.get("/customers/:customer_id/orders", function (req, res) {
+                console.log(req.body);
+                const custId = req.params.customer_id ;
+                
+                pool
+                .query("SELECT orders.order_reference,  orders.order_date, products.product_name, products.unit_price, suppliers.supplier_name , order_items.quantity  FROM orders   INNER JOIN order_items ON order_items.order_id = orders.id INNER JOIN products    ON order_items.product_id = products.id INNER JOIN suppliers   ON products.supplier_id = suppliers.id where customer_id=$1", [custId])
+                .then((result) => res.json(result.rows))
+                .catch((e) => console.error(e));
```

Comment:
Nearly correct. But `unit_price` should come from `product_availability ` table, `products` table has no such column.